lib/strategy.js

```diff
@@ -1,228 +1,58 @@
-            tokens.accessToken = accessToken;
-
-            if (params.id_token) {
-                tokens.idToken = params.id_token;
-                callback(null, tokens, refreshToken, params);
-            } else {
-                callback(null, accessToken, refreshToken, params);
-            }
+    // pass a custom Resolver, that ignores WebFinger
+    this.options.resolver = this.options.resolver || new Resolver(this.options);
+
+    // return back the Authentiq application's Client, based on the options
+    function getClientCallback(issuer, cb) {
+        return cb(null, {
+            id: options.clientID,
+            secret: options.clientSecret,
+            redirectURIs: [
+                options.callbackURL
+            ]
         });
-    }
-}
+    };
 
-util.inherits(Strategy, OAuth2Strategy);
-
-/**
- * Retrieve user profile from Authentiq ID.
- *
- * This function constructs a normalized profile, with the following properties:
- *
- *   - `provider`         always set to `authentiq`
- *   - `id`               the user's Authentiq ID
- *   - `displayName`      the user's full name
- *   - `email`            the user's shared email
- *   - `phone`            the user's shared phone
- *   - `address`          the user's shared Address
- *   - `raw`              the users raw data
- *
- * @param {string} accessToken
- * @param {function} done
- * @access protected
- */
-
-Strategy.prototype.userProfile = function (accessToken, done) {
-    if (accessToken.idToken) {
-        var profile = jwt.verify(accessToken.idToken, this.options.clientSecret);
-        done(null, profile);
-    } else {
-        this._oauth2.get(this.options.userProfileURL, accessToken.accessToken, function (err, body, res) {
-            var json;
-
-            if (err) {
-                if (err.data) {
-                    try {
-                        json = JSON.parse(err.data);
-                    } catch (_) {
-                    }
-                }
-
-                if (json && json.message) {
-                    return done(new APIError(json.message));
-                }
-                return done(new InternalOAuthError('Failed to fetch user profile', err));
-            }
-
-            try {
-                json = JSON.parse(body);
-            } catch (ex) {
-                return done(new Error('Failed to parse user profile'));
-            }
-
-            var profile = Profile.parse(json);
-            profile.provider = 'authentiq';
-            profile._raw = body;
-            profile._json = json;
-            // return the profile here and let the developer decide what to do with it
-
-            done(null, getProfile(body));
-        });
-    }
-};
+    this.options.getClientCallback = this.options.getClientCallback || getClientCallback;
 
+    OAuth2Strategy.call(this, this.options, verify);
 
-function verifyIDToken(idToken) {
-    return jwt.verify(idToken, this.options.client_secret);
+    this.name = 'authentiq';
```

Comment:
@alexkeramidas it's smart that they return an error if client_id, secret, or callback_url are not set. Maybe you can add this logic https://github.com/auth0/passport-auth0-openidconnect/blob/master/lib/passport-auth0-openidconnect/strategy.js#L25-L32